parameters.cpp

```diff
@@ -179,6 +163,27 @@ std::vector<int> P::numPasses; //numpasses
 std::string tracerString; /*!< Fieldline tracer to use for coupling ionosphere and magnetosphere */
 bool P::computeCurvature;
 
+void initParameters() {
+   meshParams.xmin = NAN;
+   meshParams.xmax = NAN;
+   meshParams.ymin = NAN;
+   meshParams.ymax = NAN;
+   meshParams.zmin = NAN;
+   meshParams.zmax = NAN;
+   meshParams.dx_ini = NAN;
+   meshParams.dy_ini = NAN;
+   meshParams.dz_ini = NAN;
+
+   meshParams.xcells_ini = numeric_limits<uint>::max();
```

Comment:
so naming issue here: meshParams can contain the actual mesh descriptions, but the final 5 values below are Fieldsolver Parameters, not mesh parameters.

One solution would be to retain the original P::xmin etc parameter types, and in addition to that, rename the meshParams you created to FSparams, and have that include the FSgrid parameters (so dx_ini * 2^-maxreflevel etc) as well as the fieldsolver parameters.